javaslang/src/test/java/javaslang/collection/AbstractSeqTest.java

```diff
@@ -1217,6 +1217,27 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
         assertThat(ofAll(of("a", "b", "c")).reverseIterator().toList()).isEqualTo(Iterator.of("c", "b", "a").toList());
     }
 
+    // -- shuffle
+
+    @Test
+    public void shouldShuffleEmpty() {
+        assertThat(empty().shuffle().isEmpty());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameLength() {
+        assertThat(of(1, 2, 3).shuffle().size()).isEqualTo(of(1, 2, 3).size());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameElements() {
+        Seq<Integer> shuffled = of(1, 2, 3).shuffle();
+        assertThat(shuffled.indexOf(1)).isNotEqualTo(-1);
```

Comment:
I understand your motive, and I think `containsAll` is still primitive enough and would signal the intent a lot better to the users than separate `indexOfs, which don't signal directly that we're testing whether it has the same elements.

However, `indexOf` and `contains` may be problematic in case of duplicates, so the cleanest way would be to sort-and-compare, i.e.
```java
final Seq<Integer> original = of(1, 2, 3);
final Seq<Integer> shuffled = original.shuffle();
assertThat(original.sorted()).isEqualTo(shuffled.sorted());
```